candle-core/src/metal_backend.rs

```diff
@@ -70,16 +112,107 @@ impl MetalDevice {
         &self.device
     }
 
-    pub fn new_buffer(&self, element_count: usize, dtype: DType) -> Buffer {
+    pub fn new_buffer(&self, element_count: usize, dtype: DType) -> Arc<Buffer> {
         let size = (element_count * dtype.size_in_bytes()) as NSUInteger;
-        self.device
-            .new_buffer(size, MTLResourceOptions::StorageModeManaged)
+        self._new_buffer(size, MTLResourceOptions::StorageModePrivate)
+    }
+
+    fn _new_buffer(&self, size: NSUInteger, option: MTLResourceOptions) -> Arc<Buffer> {
```

Comment:
The leading `_` typically are used to bypass the "is used" warnings. For internal functions, naming them with an underscore at the end seems usually better.

Reply:
I renamed this to `allocate_buffer` which is actually a better name imo.
I wrote `_new_buffer` by lack of good name in the meantime.